rag_experiment_accelerator/run/index.py

```diff
@@ -33,48 +33,47 @@ def run(
     index_config: IndexConfig,
     file_paths: list[str],
     mlflow_client: mlflow.MlflowClient,
-) -> dict[str]:
+) -> str:
     """
     Runs the main experiment loop, which chunks and uploads data to Azure AI Search indexes based on the configuration specified in the Config class.
 
     Returns:
-        Index dictionary containing the names of the indexes created.
+        The name of the index created.
     """
     pre_process = Preprocess(True)
-    index_dict = {"indexes": []}
 
-    with TimeTook(
-        f"create Azure Search Index {index_config.index_name()}", logger=logger
-    ):
+    index_name = index_config.index_name()
+    with TimeTook(f"create Azure Search Index {index_name}", logger=logger):
         create_acs_index(
             environment.azure_search_service_endpoint,
-            index_config.index_name(),
+            index_name,
             environment.azure_search_admin_key,
-            index_config.embedding_model.dimension,
+            config.get_embedding_model(
```

Comment:
Look, I've just checked the code and it seems like (maybe there is a chance that it happens before, but I might have missed it) that here during model creation code will download sentence-transformer model from the internet.
This will screw up TimeTook.
To avoid getting slapped this way, it's a good idea to get this dimension outside of TimeTook so downloading of the model is not part of time eval

Reply:
Downloading happens once - on config creation (method is create_embedding_model). And this method is just getting it from a dictionary of models.